main.go

```diff
@@ -14,6 +15,82 @@ import (
 
 const Version = "v0.4"
 
+type EC2InstanceType struct {
+	Name                 string
+	MaximumCredits       float64
+	CreditsEarnedPerHour int
+}
+
+func getInstanceId() (string, error) {
+	rep, err := http.Get("http://instance-data/latest/meta-data/instance-id")
+	if err != nil {
+		return "", err
+	}
+
+	defer rep.Body.Close()
+	body, err := ioutil.ReadAll(rep.Body)
+	if err != nil {
+		return "", err
+	}
+
+	instanceId := string(body)
+	return instanceId, nil
```

Comment:
I'd condense these two lines (ie no point defining a variable, when it's just used once, briefly)